test/bigml/test/api/source.clj

```diff
@@ -8,11 +8,13 @@
   (:use clojure.test))
 
 (defn- create-and-test [artifact & params]
-  (let [initial (api/get-final (apply source/create artifact params))
+  (let [{:keys [username api_key dev_mode] :as sticky-pars} params
```

Comment:
> On 14 Jan 2016, at 18:20, Adam Ashenfelter <notifications@github.com> wrote:
> 
> In test/bigml/test/api/source.clj:
> 
> > @@ -8,11 +8,13 @@
> >    (:use clojure.test))
> >  
> >  (defn- create-and-test [artifact & params]
> > -  (let [initial (api/get-final (apply source/create artifact params))
> > +  (let [{:keys [username api_key dev_mode] :as sticky-pars} params
> 
> You can use the :as to bind params as a hash-map in the fn definition, which makes using select-keys a little less messy. Like so:

It seems that this would be a way to use your hint:

```
(defn- create-and-test [artifact & {:as params}]
  (let [sticky-pars (flatten (seq (select-keys 
                                   params
                                   [:username :api_key :dev_mode])))
        params (flatten (seq params))
        …
```

btw, I rewrote the remote history, I hope this is not trouble since review happens on github...